### PR TITLE
Add type hints to urllib3.util.request

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: src/urllib3/util/request.py [hunk 1]
```diff
@@ -66,7 +67,7 @@ def make_headers(
         >>> make_headers(accept_encoding=True)
         {'accept-encoding': 'gzip,deflate'}
     """
-    headers = {}
+    headers: Dict[str, Any] = {}
```
Comment (sethmlarson) [inline src/urllib3/util/request.py:70]: Why is the value type `Any` instead of `str`?

File: src/urllib3/util/request.py [hunk 2]
```diff
@@ -98,7 +99,9 @@ def make_headers(
     return headers
 
 
-def set_file_position(body, pos):
+def set_file_position(
+    body: Union[BinaryIO, TextIO], pos: Optional[Union[int, object]]
```
Comment (sethmlarson) [inline src/urllib3/util/request.py:103]: Is there any advantage to using `Union[BinaryIO, TextIO]` over `IO`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Two small comments then this looks good :)

File: src/urllib3/util/request.py (hunk 1 above)
Comment (hramezani) [reply src/urllib3/util/request.py:70]: I was wrong. `str` is the right type.

File: src/urllib3/util/request.py (hunk 2 above)
Comment (hramezani) [reply src/urllib3/util/request.py]: Just to make it clear. I've replaced it with `IO[AnyStr]`

**Review by sethmlarson — DISMISSED** [review]
LGTM

**Review by sethmlarson — APPROVED** [review]

**Pull request merged**